site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/FileTable.js

```diff
@@ -0,0 +1,82 @@
+import _get from "lodash/get";
+import PropTypes from "prop-types";
+import React from "react";
+import {
+    Icon,
+    Popup,
+    Table,
+} from "semantic-ui-react";
+import { humanReadableBytes } from "react-invenio-deposit";
+
+
+const FileTableRow = ({ handleDelete, file, decimalSizeDisplay }) => {
+    return (
+        <Table.Row key={file.name}>
+            <Table.Cell data-label="Filename" width={10}>
+                <>
+                    <label>{file.name}</label>
+                </>
+            </Table.Cell>
+            <Table.Cell data-label="Size" width={2}>
+                {file.size ? humanReadableBytes(file.size, decimalSizeDisplay) : ""}
+            </Table.Cell>
+
+            <Table.Cell textAlign="right" width={2}>
+                {
+                    (
+                        <Icon
+                            link
+                            className="action primary"
+                            name="trash alternate outline"
+                            onClick={() => handleDelete(file)}
+                            aria-label={"Delete file"}
+                            title={"Delete file"}
+                        />
+                    )}
+            </Table.Cell>
+
+        </Table.Row>
+    )
+};
+
+FileTableRow.propTypes = {
+    handleDelete: PropTypes.func.isRequired,
+    file: PropTypes.object.isRequired,
+    decimalSizeDisplay: PropTypes.bool
+};
+
+FileTableRow.defaultProps = {
+    decimalSizeDisplay: false
+}
+
+const FileTable = ({ filesList, handleDelete, decimalSizeDisplay }) => {
+    return (
+        <Table>
+            <Table.Body>
+                {filesList.map((file) => {
+                    return (
+                        <FileTableRow
```

Comment:
minor: I would either split the `FileTableRow` in a new file or integrate the render method directly here.

Reply:
Integrated the render method directly